src/communication/network_interface_sr.c

```diff
@@ -7367,28 +7381,55 @@ smmon_get_module_info (THREAD_ENTRY * thread_p, unsigned int rid, char *request,
   // *INDENT-OFF*
   for (const auto &m_info : module_info)
     {
-      size += or_packed_string_length (m_info.name, NULL);
+      string_len = or_packed_string_length (m_info.name, NULL);
+      if (string_len % MAX_ALIGNMENT)
+        {
+          size += string_len + OR_INT_SIZE;
+        }
+      else
+        {
+          size += string_len;
+        }
+      //size += or_packed_string_length (m_info.name, NULL);
       // Size of stat information (OR_INT64_SIZE * 3 + OR_INT_SIZE)
       // and size of num_comp (OR_INT_SIZE)
       size += OR_INT64_SIZE * 3 + OR_INT_SIZE * 2;
 
       for (const auto &comp_info : m_info.comp_info)
         {
-          size += or_packed_string_length (comp_info.name, NULL);
+          string_len = or_packed_string_length (comp_info.name, NULL);
```

Comment:
Aligning should be done based on `size`, or we would get the same error when size is not aligned with 8 bytes.

```suggestion
          size +=  or_packed_string_length (comp_info.name, NULL);
          size = size % MAX_ALIGHMENT ? size + OR_INT_SIZE;
```

Reply:
I guess `size + OR_MAX_ALIGHMENT_SIZE;` is safer.

Reply:
Done.